textmap/vectorizers.py

```diff
@@ -679,69 +676,291 @@ def change_basis(self, X, column_index_dictionary):
+        # creates a bag of words representation of that document.
+        self.word_cooccurrence_vectorizer_ = create_processing_pipeline_stage(
+            self.word_cooccurrence_vectorizer,
+            _COOCCURRENCE_VECTORIZERS,
+            self.word_cooccurrence_vectorizer_kwds,
+            "Word CooccurrenceVectorizer",
+        )
+        if self.word_cooccurrence_vectorizer_ is not None:
+            self.representation_words_ = self.word_cooccurrence_vectorizer_.fit_transform(
+                tokens_by_sentence
+            )
+            if self.feature_basis_transformer_ is not None:
+                self.representation_words_ = self.feature_basis_transformer_.change_basis(
+                    self.representation_words_,
+                    self.word_cooccurrence_vectorizer_.column_index_dictionary_,
+                )
+
+        # DOCUMENT VECTORIZER
+        # This should be the same column representation oas the word cooccurrence vectorizer above.
+        # By default this is a bag of words representation.
+        self.doc_vectorizer_kwds_ = initialize_kwds(
+            self.doc_vectorizer_kwds, {"fit_unique": self.fit_unique}
+        )
+        self.doc_vectorizer_ = create_processing_pipeline_stage(
+            self.doc_vectorizer,
+            _DOCUMENT_VECTORIZERS,
+            self.doc_vectorizer_kwds_,
+            "DocVectorizer",
+        )
+        if self.doc_vectorizer_ is not None:
+            self.representation_docs_ = self.doc_vectorizer_.fit_transform(
+                tokens_by_document
+            )
+            if self.feature_basis_transformer_ is not None:
+                self.representation_docs_ = self.feature_basis_transformer_.change_basis(
+                    self.representation_docs_,
+                    self.doc_vectorizer_.column_index_dictionary_,
+                )
+
+        # Putting docs above words.  Docs are often referenced solely by their index while words have a better
+        # token label.  This way docs keep their index.
+        if isinstance(self.representation_docs_, sparse.csr.csr_matrix):
+            vstack = sparse.vstack
+        elif isinstance(self.representation_docs_, np.ndarray):
+            vstack = np.vstack
+        else:
+            raise ValueError(
+                f"Your representation must be a numpy array or sparse matrix;"
+                f"representation_docs_ is of type {type(self.representation_docs_)}"
+            )
+
+        self.representation_ = vstack(
+            [self.representation_docs_, self.representation_words_]
+        )
+
+        # Promote a bunch of dictionaries
+        # Ugh, this will currently break horribly if you set lots of steps to None.
+        # We can either prevent them from doing that for the feature_basis_transformer and doc_vectorizer or...
+        # This is returned in the order that the tokens occur as rows in the represenation_words_
+        self.vocabulary_ = self.doc_vectorizer_.vocabulary_
```

Comment:
One way to do this is as follows:
```suggestion
        self.vocabulary_ = self.doc_vectorizer_.vocabulary_ if self.doc_vectorizer_ else None
```
This will set the vocabulary to ``None`` is ``self.doc_vectorizer_`` is ``None``, otherwise it will use ``self.doc_vectorizer_.vocabulary_``.